Chore/fix workspace build again

NSeydoux opened the pull request.

NSeydoux committed:
Use node-resolve rollup plugin

NSeydoux committed:
Run mock release in CI to check

NSeydoux committed:
Fix build order

This removes using rollup node-resolve plugin and typescript project references, all that was needed was to set the build order in the workspaces field for the compilation to suceed.

NSeydoux committed:
fixup! Fix build order

NSeydoux committed:
Cleanup mock release

ThisIsMissEm commented on package.json:
```diff
@@ -10,7 +10,10 @@
+    "packages/internal-test-env",
+    "packages/internal-playwright-testids",
+    "packages/internal-playwright-helpers",
+    "packages/jest-jsdom-polyfills"
```
This change shouldn't be needed, this just tells npm to glob that directory..

NSeydoux replied:
I'll try what you suggested in your comment and have lerna run the build, because I actually thought that was already the case and that lerna was hooking into the workspaces setup. This changes just adds an order in which the packages are built, because using alphabetical order results in `packages/internal-playwright-helpers` being built first, whil it has dependencies on the other packages.

ThisIsMissEm reviewed (commented):
I'd like to better understand why this is failing often.

ThisIsMissEm commented:
If you're seeing `npm run build --workspaces --if-present` fail due to dependencies between packages, then the answer will be to use lerna to run the build instead, as npm workspaces are topologically sorted, iirc.

ThisIsMissEm commented:
Aside, it can't hurt to add a ci.yml which always runs a build? Might be necessary now that we're getting more complex packages here

NSeydoux committed:
Revert "Cleanup mock release"

This reverts commit 2af376359e97146dea4ae953627e9e2dcfdb3b80.

NSeydoux committed:
Use Lerna to run build

Building based on workspaces alone (i.e. `npm run build --workspaces`) doesn't account for dependencies between workspaces, resulting in an arbitrary build order being based on the globbing pattern/order in which workspaces are defined. Instead, this commit uses Lerna to run the build, and relies on Nx to figure out the build order based on the dependency graph.

NSeydoux committed:
Add CI to verify building is passing

chelseapinka commented on lerna.json:
```diff
@@ -1,5 +1,5 @@
 {
-  "useNx": false,
+  "useNx": true,
```
Reading this page, it all makes sense the order of operations being taken care of by lerna here with help from Nx to see it that it cannot build the helpers package before testids. https://nx.dev/recipes/adopting-nx/lerna-and-nx Makes sense :+1:

NSeydoux committed:
Remove building on node 14

ThisIsMissEm commented on .github/workflows/release.yml:
```diff
@@ -20,7 +20,11 @@ jobs:
+      # the other way around results in top-level dependencies (namely, lerna)
+      # not being installed.
+      - run: npm ci
+      - run: npm run build
```
Release runs build as part of it, or, at least should. Part of publishing a package is running the build step iirc
